rustls-post-quantum/src/mlkem.rs

```diff
@@ -0,0 +1,96 @@
+use aws_lc_rs::kem;
+use aws_lc_rs::unstable::kem::ML_KEM_768;
+use rustls::crypto::{ActiveKeyExchange, CompletedKeyExchange, SharedSecret, SupportedKxGroup};
+use rustls::ffdhe_groups::FfdheGroup;
+use rustls::{Error, NamedGroup, PeerMisbehaved, ProtocolVersion};
+
+/// This is the [MLKEM] key exchange.
+///
+/// [MLKEM]: https://datatracker.ietf.org/doc/draft-connolly-tls-mlkem-key-agreement
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Debug)]
+pub struct MLKEM768;
+
+impl SupportedKxGroup for MLKEM768 {
+    fn start(&self) -> Result<Box<dyn ActiveKeyExchange>, Error> {
+        let decaps_key = kem::DecapsulationKey::generate(&ML_KEM_768)
+            .map_err(|_| Error::FailedToGetRandomBytes)?;
+
+        let encaps_key = decaps_key
+            .encapsulation_key()
+            .map_err(|_| Error::FailedToGetRandomBytes)?;
+
+        let pub_key_bytes = encaps_key.key_bytes().unwrap();
```

Comment:
This is documented as returning an unspecified error for "Any failure to retrieve the EnscapsulationKey bytes.". It's not immediate obvious to me what sort of condition could cause that failure.

Can you add a justification comment for why the panic is unreachable, or should it be mapped to `Error::Other` or `Error::General`?